Code/Framework/AzToolsFramework/Tests/FocusMode/EditorFocusModeTests.cpp

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzTest/AzTest.h>
+#include <AzCore/UserSettings/UserSettingsComponent.h>
+#include <AzCore/Component/TransformBus.h>
+#include <AzToolsFramework/FocusMode/FocusModeInterface.h>
+#include <AzToolsFramework/UnitTest/AzToolsFrameworkTestHelpers.h>
+
+namespace AzToolsFramework
+{
+    class EditorFocusModeTests
+        : public ::testing::Test
+    {
+    protected:
+        void SetUp() override
+        {
+            m_app.Start(m_descriptor);
+
+            // Without this, the user settings component would attempt to save on finalize/shutdown. Since the file is
+            // shared across the whole engine, if multiple tests are run in parallel, the saving could cause a crash 
+            // in the unit tests.
+            AZ::UserSettingsComponentRequestBus::Broadcast(&AZ::UserSettingsComponentRequests::DisableSaveOnFinalize);
+
+            GenerateTestHierarchy();
+        }
+
+        void GenerateTestHierarchy() 
+        {
+            /*
+            *   City
+            *   |_  Street
+            *       |_  Car
+            *       |   |_ Passenger
+            *       |_  SportsCar
+            *           |_ Passenger
+            */
+
+            m_entityMap["cityId"] =         CreateEditorEntity("City",      AZ::EntityId());
+            m_entityMap["streetId"] =       CreateEditorEntity("Street",    m_entityMap["cityId"]);
+            m_entityMap["carId"] =          CreateEditorEntity("Car",       m_entityMap["streetId"]);
+            m_entityMap["passengerId1"] =   CreateEditorEntity("Passenger", m_entityMap["carId"]);
+            m_entityMap["sportsCarId"] =    CreateEditorEntity("SportsCar", m_entityMap["streetId"]);
+            m_entityMap["passengerId2"] =   CreateEditorEntity("Passenger", m_entityMap["sportsCarId"]);
```

Comment:
suggestion: It might be worth moving some of these strings to constants so it's easier to keep track of changes to them (especially if they're referenced later)